File: php/iOSAppTest.php
<?php
require_once('PHPUnit/Extensions/AppiumTestCase.php');
include("configs.php");

class IOSAppTest extends PHPUnit_Extensions_AppiumTestCase {
  public $firstQuestion = 'Acura MDX';
  public $secondQuestion = 'Cruise Control';

  public static $browsers = desired_caps_ios_app;

  public function test_ios_app() {
    $this->search_questions_on_Acura_Support_Community();
    $this->search_IFixit_on_home_screen();
  }

  public function search_questions_on_Acura_Support_Community() {
    print 'should allow to navigate to some devices on Acura Support Community';
   
   /* Steps: 
    1. Click on "Car and Truck" Categories on Homepage
    2. Click on "Acura" Categories
    Expected: 
    1. General Information is "Acura".
    2.Verify three devices below displays. 
    + 1994-2001 Acura Integra 
    + 1996-1998 Acura TL 
    + 2004 - 2008 Acura TSX
    */
   
    $this->byXpath("//XCUIElementTypeButton[@name='START A REPAIR']")->click();
    sleep(5);
    $this->byXpath("//*[@name='Car and Truck']")->click();
    sleep(2);
    $this->byXpath("//*[@name='Acura']")->click();
    sleep(2);

    $acuraText = $this->byXpath("//XCUIElementTypeNavigationBar")->attribute('name');
    $has_AcuraIntegra= $this->byXpath("//XCUIElementTypeStaticText[@name='Acura Integra']")->displayed();
    $has_AcuraMDX = $this->byXpath("//XCUIElementTypeStaticText[@name='Acura MDX']")->displayed();
    $has_AcuraRL = $this->byXpath("//XCUIElementTypeStaticText[@name='Acura RL']")->displayed();
    $has_AcuraTL = $this->byXpath("//XCUIElementTypeStaticText[@name='Acura TL']")->displayed();
    $has_AcuraTSX = $this->byXpath("//XCUIElementTypeStaticText[@name='Acura TSX']")->displayed();
  
    $this->assertEquals('Acura', $acuraText);
    $this->assertTrue($has_AcuraIntegra);
    $this->assertTrue($has_AcuraMDX);
    $this->assertTrue($has_AcuraRL);
    $this->assertTrue($has_AcuraTL);
    $this->assertTrue($has_AcuraTSX);
  }

  public function search_IFixit_on_home_screen() {
    print 'should allow to search iFixit on Home screen';

   /*
    Steps:
    1. Reopen iFixit app
    2. Click Search menu on the left menu bar
    3. Search keyword 'Macbook Pro 2015'
    4. Press Enter button on keyboard
    Expected: It should show at least 33 results.
    5. Clear the current content
    6. Search keyword 'Acura' on Categories tab
    Expected: It should show at least 6 results.
    */

    $this->launchApp();
    sleep(5);
    $this->byXpath("//XCUIElementTypeButton[@name='START A REPAIR']")->click();
    sleep(2);
    $this->byXpath("//*[@name='Search']")->click();
    sleep(2);
    $this->byXpath("//XCUIElementTypeSearchField[@name='Search']")->click();
    $this->keys('Macbook Pro 2015');
    sleep(2);
    $first_list_result = $this->elements($this->using('xpath')->value("//XCUIElementTypeStaticText[contains(@label,'MacBook Pro')]"));

    $this->byXpath("//XCUIElementTypeButton[@name='Cancel']")->click();
    sleep(2);
    $this->byXpath("//*[@name='Search']")->click();
    sleep(2);
    $this->byXpath("//XCUIElementTypeSearchField[@name='Search']")->click();
    sleep(2);
    $this->keys('Acura');
    sleep(5);
    $this->byXpath("//XCUIElementTypeButton[@name='Categories']")->click();
    sleep(2);
    $second_list_result = $this->elements($this->using('xpath')->value("//XCUIElementTypeStaticText[contains(@label,'Acura')]"));

    $this->assertGreaterThanOrEqual(33, count($first_list_result), 'The expected results are greater or equal to 33 results.');
    $this->assertGreaterThanOrEqual(6, count($second_list_result), 'The expected results are greater or equal to 6 results.');
  }
}

File: java/java_testng_junit/src/main/java/kobiton/com/testng/iOSAppTest.java
package kobiton.com.testng;
import java.util.List;
import java.util.concurrent.TimeUnit;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterTest;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;
import io.appium.java_client.ios.IOSDriver;
import kobiton.com.configs.Configs;
import com.google.common.collect.ImmutableMap;

public class iOSAppTest {
	
	public static IOSDriver<WebElement> driver = null;
			
	@BeforeTest
    public void Setup() {
		driver = new IOSDriver<WebElement>(Configs.kobitonServerUrl(), Configs.desiredCapabilitiesiOSApp());
		driver.manage().timeouts().implicitlyWait(120, TimeUnit.SECONDS);
    }
	
   @AfterTest
    public void Teardown() {
      try {
      	if(driver != null)
          driver.quit();
      } catch (Exception e) {
          e.printStackTrace();
      }
    }
   
   @Test(priority = 1, description="should allow to navigate to some devices on Acura Support Community")
   public void testNavigationOnAcuraSupportCommunity() {
	   /*
		 * Steps: 
		 * 1. Click on "Car and Truck" Categories on Homepage 
		 * 2. Click on "Acura" Categories
		 * 
		 * Expected: 
		 * 1. General Information is "Acura".
		 * 2.Verify five devices below displays.
		 * + Acura Integra 
		 * + Acura MDX
		 * + Acura RL 
		 * + Acura TL 
		 * + Acura TSX
		 */
	   
	  driver.findElementByXPath("//XCUIElementTypeButton[@name='START A REPAIR']").click();
		sleep(2);
		driver.findElementByXPath("//*[@name='Car and Truck']").click();
		driver.findElementByXPath("//*[@name='Acura']").click();
		sleep(2);
		(new WebDriverWait(driver, 60)).until(
			ExpectedConditions.elementToBeClickable(By.xpath("//XCUIElementTypeNavigationBar")));
		
		String acuraText = driver
			.findElementByXPath("//XCUIElementTypeNavigationBar").getAttribute("name");
		boolean hasAcuraIntegra = driver
			.findElementByXPath("//XCUIElementTypeStaticText[@name='Acura Integra']").isDisplayed();
		boolean hasAcuraMDX = driver
			.findElementByXPath("//XCUIElementTypeStaticText[@name='Acura MDX']").isDisplayed();
		boolean hasAcuraRL = driver
			.findElementByXPath("//XCUIElementTypeStaticText[@name='Acura RL']").isDisplayed();
		boolean hasAcuraTL = driver
			.findElementByXPath("//XCUIElementTypeStaticText[@name='Acura TL']").isDisplayed();
		boolean hasAcuraTSX = driver
			.findElementByXPath("//XCUIElementTypeStaticText[@name='Acura TSX']").isDisplayed();

		driver.closeApp();
		
		Assert.assertEquals(acuraText, "Acura");
		Assert.assertEquals(hasAcuraIntegra, true);
		Assert.assertEquals(hasAcuraMDX, true);
		Assert.assertEquals(hasAcuraRL, true);
		Assert.assertEquals(hasAcuraTL, true);
		Assert.assertEquals(hasAcuraTSX, true);
   }
   
   @Test(priority = 2, description="should allow to search iFixit on Home screen")
   public void testSearchIFixitOnHomeScreen() {
	   /*
	    Steps:
	    1. Reopen iFixit app
	    2. Click Search menu on the left menu bar
	    3. Search keyword 'Macbook Pro 2015'
	    4. Press Enter button on keyboard

	    Expected: It should show at least 33 results.

	    5. Clear the current content
	    6. Search keyword 'Acura' on Categories tab

	    Expected: It should show at least 6 results.
	    */
	   
	   driver.launchApp();
	   driver.findElementByXPath("//XCUIElementTypeButton[@name='START A REPAIR']").click();
	   sleep(2);
	   driver.findElementByXPath("//*[@name='Search']").click();
	   driver.findElementByXPath("//XCUIElementTypeSearchField[@name='Search']").sendKeys("Macbook Pro 2015");
	   sleep(2);
	   
	   List<WebElement> firstResult = driver.findElementsByXPath("//XCUIElementTypeStaticText[contains(@label,'MacBook Pro')]");
	   
	   driver.findElementByXPath("//XCUIElementTypeButton[@name='Cancel']").click();
	   driver.findElementByXPath("//*[@name='Search']").click();
	   driver.findElementByXPath("//XCUIElementTypeSearchField[@name='Search']").sendKeys("Acura");
	   driver.findElementByXPath("//XCUIElementTypeButton[@name='Categories']").click();
	   sleep(2);
	   
	   List<WebElement> secondResult = driver.findElementsByXPath("//XCUIElementTypeStaticText[contains(@label,'Acura')]");
	   driver.executeScript("mobile: scroll", ImmutableMap.of("direction", "down"));
	   driver.executeScript("mobile: scroll", ImmutableMap.of("direction", "up"));
	   
		Assert.assertTrue(firstResult.size() >= 33,
			"The expected results are greater or equal to 33 results.");
		
		Assert.assertTrue(secondResult.size() >= 6,
			"The expected results are greater or equal to 6 results.");
   }
   
   public void sleep(int seconds) {
		try {
			Thread.sleep(seconds * 1000);
		} catch (InterruptedException e) {
			e.printStackTrace();
		}
	}
}
